dotnet/src/Connectors/Connectors.Memory.InMemory/InMemoryVectorStore.cs

```diff
@@ -27,30 +31,18 @@ public sealed class InMemoryVectorStore : IVectorStore
     /// <summary>
     /// Initializes a new instance of the <see cref="InMemoryVectorStore"/> class.
     /// </summary>
-    public InMemoryVectorStore()
+    /// <param name="options">Optional configuration options for this class</param>
+    public InMemoryVectorStore(InMemoryVectorStoreOptions? options = default)
     {
-        this._internalCollections = new();
+        this._options = options ?? new InMemoryVectorStoreOptions();
+        this._internalCollections = this._options.StoreName is null ? new() : s_stores.GetOrAdd(this._options.StoreName, _ => new());
```

Comment:
This method can add something to a static cache, but which method is responsible for removing it? And when?

Reply:
Note: I ended up removing this feature altogether, following discussion with Adam.